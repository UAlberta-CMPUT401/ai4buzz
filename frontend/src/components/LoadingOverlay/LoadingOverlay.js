import React from 'react';
import Spinner from './spinner.svg';

import styles from './LoadingOverlay.module.css';

const LoadingOverlay = () => {
  return (
    <div className={styles.loadingOverlay}>
      <object type="image/svg+xml" data={Spinner}>
        svg-animation
      </object>
    </div>
  );
};

export default LoadingOverlay;
