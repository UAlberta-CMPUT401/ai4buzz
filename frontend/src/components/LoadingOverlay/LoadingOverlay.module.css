.loadingOverlay {
  position: absolute;
  width: 100%;
  height: 100%;
  display: grid;
  place-items: center;
  background-color: rgba(0, 0, 0, 0.5);
}
