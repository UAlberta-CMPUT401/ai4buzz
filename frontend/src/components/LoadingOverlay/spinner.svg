<?xml version="1.0" encoding="utf-8"?>
<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" style="margin: auto; background: none; display: block; shape-rendering: auto;" width="200px" height="200px" viewBox="0 0 100 100" preserveAspectRatio="xMidYMid">
<circle cx="84" cy="50" r="10" fill="#ffffff">
    <animate attributeName="r" repeatCount="indefinite" dur="0.390625s" calcMode="spline" keyTimes="0;1" values="14;0" keySplines="0 0.5 0.5 1" begin="0s"></animate>
    <animate attributeName="fill" repeatCount="indefinite" dur="1.5625s" calcMode="discrete" keyTimes="0;0.25;0.5;0.75;1" values="#ffffff;#00edff;#a52bff;#ff6ed0;#ffffff" begin="0s"></animate>
</circle><circle cx="16" cy="50" r="10" fill="#ffffff">
  <animate attributeName="r" repeatCount="indefinite" dur="1.5625s" calcMode="spline" keyTimes="0;0.25;0.5;0.75;1" values="0;0;14;14;14" keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1" begin="0s"></animate>
  <animate attributeName="cx" repeatCount="indefinite" dur="1.5625s" calcMode="spline" keyTimes="0;0.25;0.5;0.75;1" values="16;16;16;50;84" keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1" begin="0s"></animate>
</circle><circle cx="50" cy="50" r="10" fill="#ff6ed0">
  <animate attributeName="r" repeatCount="indefinite" dur="1.5625s" calcMode="spline" keyTimes="0;0.25;0.5;0.75;1" values="0;0;14;14;14" keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1" begin="-0.390625s"></animate>
  <animate attributeName="cx" repeatCount="indefinite" dur="1.5625s" calcMode="spline" keyTimes="0;0.25;0.5;0.75;1" values="16;16;16;50;84" keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1" begin="-0.390625s"></animate>
</circle><circle cx="84" cy="50" r="10" fill="#a52bff">
  <animate attributeName="r" repeatCount="indefinite" dur="1.5625s" calcMode="spline" keyTimes="0;0.25;0.5;0.75;1" values="0;0;14;14;14" keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1" begin="-0.78125s"></animate>
  <animate attributeName="cx" repeatCount="indefinite" dur="1.5625s" calcMode="spline" keyTimes="0;0.25;0.5;0.75;1" values="16;16;16;50;84" keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1" begin="-0.78125s"></animate>
</circle><circle cx="16" cy="50" r="10" fill="#00edff">
  <animate attributeName="r" repeatCount="indefinite" dur="1.5625s" calcMode="spline" keyTimes="0;0.25;0.5;0.75;1" values="0;0;14;14;14" keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1" begin="-1.171875s"></animate>
  <animate attributeName="cx" repeatCount="indefinite" dur="1.5625s" calcMode="spline" keyTimes="0;0.25;0.5;0.75;1" values="16;16;16;50;84" keySplines="0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1;0 0.5 0.5 1" begin="-1.171875s"></animate>
</circle>
<!-- [ldio] generated by https://loading.io/ --></svg>