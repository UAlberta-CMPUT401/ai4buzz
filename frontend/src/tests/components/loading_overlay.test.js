import React from 'react';
import LoadingOverlay from '../../components/LoadingOverlay/LoadingOverlay';
import renderer from 'react-test-renderer';

it('Renders loading overlay correctly', () => {
  const tree = renderer.create(<LoadingOverlay />).toJSON();
  expect(tree).toMatchSnapshot();
});
