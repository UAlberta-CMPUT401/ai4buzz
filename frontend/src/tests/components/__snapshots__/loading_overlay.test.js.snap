// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders loading overlay correctly 1`] = `
<div
  className="loadingOverlay"
>
  <div>
    <object
      data="spinner.svg"
      type="image/svg+xml"
    >
      svg-animation
    </object>
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      This might take a bit...
    </div>
  </div>
</div>
`;
